cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/partitioned/ConsumablePartition.java

```diff
@@ -54,7 +54,7 @@
   void take();
 
   /**
-   * Marks the ProcessState as AVAILABLE, resets the timestamp to 0, and increments the number of failures by 1
+   * Marks the ProcessState as AVAILABLE and resets the timestamp to 0
```

Comment:
I also think that the naming of the method sort of implies that it has failed. Whereas the situation where this is used is when the client decides not put it back without having tried to process it. You could name the new method `untake` or `yield` instead of `retryWithoutFailureCount`